integrated_channels/blackboard/client.py

```diff
@@ -37,7 +42,43 @@ def __init__(self, enterprise_configuration):
         self.expires_at = None
 
     def create_content_metadata(self, serialized_data):
-        """TODO"""
+        """
+        Create a course from serialized course metadata
+        """
+        channel_metadata_item = json.loads(serialized_data.decode('utf-8'))
+        if 'externalId' not in channel_metadata_item:
+            raise ClientError("No externalId found in metadata, please check json data format", 400)
+
+        external_id = channel_metadata_item['externalId']
+
+        # blackboard does not support all characters in our courseIds so let's gen a hash instead
+        course_id_generated = abs(hash(external_id))
+
+        copy_of_channel_metadata = copy.deepcopy(channel_metadata_item)
+        copy_of_channel_metadata['courseId'] = course_id_generated
+
+        serialized_channel_metadata = json.dumps(copy_of_channel_metadata)
+
+        LOGGER.info("Transmitting create_content_metadata with course_id: %s, %s",
+                    course_id_generated, serialized_channel_metadata)
+        self._create_session()
+        create_course_url = "{}/learn/api/public/v3/courses".format(
+            self.enterprise_configuration.blackboard_base_url
+        )
+        status_code, response_text = self._post(create_course_url, serialized_channel_metadata)
```

Comment:
A good practice that I like from Talia's work is wrapping the requests/error handling (checking status code). I added helper methods for posts, gets and patches in my PR if you want to use those!

Reply:
hmm, not clear on how to do that, where are your helpers? I only have helpers here for oauth and auth header creation

Reply:
I think I see yours now https://github.com/edx/edx-enterprise/pull/998/files#diff-e8224b0f5aaaee88283c12d5d533f5700f33f2e63c7939e93f814ef94bf95716R122 yeah I guess we could just have error handling inside the _put/_post methods. Typically I don't consider it a great idea to have automatic error handling in lower level methods since 400 is an issue to be dealt with by whoever calls _put/_post etc while the lower level put just does the putting.

Also if we have error handling in _put/_post we don't have the ability to customize error messages  (unless we catch and rethrow?)

Somehow I will find a way to leverage your code once it merges while also being able to throw exceptions with custom messages

Reply:
also noting thatt your code does return the response_text along with the code so it's actually going to be quite good I thin even as is and I can try not even having any error handling in my code initially then see if I need to tune messages